### DISCUSSION TITLE
Tasmota flash to sonoff S20

### DISCUSSION THREAD

**Discussion by CookiesForevaaaaa** [arendst/Tasmota#15692] [Q&A] [answered]
Hey guys,

I am trying to flash tasmota to my sonoff S20 switch. I have connected it with my FTDI to computer and everything seems fine. RX to TX and other way around. 3V3 works fine I guess since a green light turns on my tsonoff switch. When I try to flash it just doesn't connect with the sonoff and flash fails. 

Any ideas?

**Comment by sfromis** [comment, marked as answer]
You did not mention connecting ground too?

To put it into flashing mode, you should hold the button (connecting gpio0 to ground) pushed while applying power, typically plugging in USB.

Note that anything claiming to be "FTDI" is usually clones so low in power supply ability that flashing is not always reliable, but I'd not expect this to be the cause with the Sonoff S20.

**Reply by CookiesForevaaaaa** [reply]
thanks for you reply, of course I connected the ground too but I did not hold the button, I hope this is the issue, will try tomorrow and keep you posted! thank you

**Reply by CookiesForevaaaaa** [reply]
it worked! flashed succesfully
